src/Chapter4.hs

```diff
@@ -629,7 +660,8 @@ Can you implement a monad version of AND, polymorphic over any monad?
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+-- andM m1 m2 = pure (&&) <*> m1 <*> m2 -- (Without >>= and anonymous function)
```

Comment:
The ordinary `&&` if lazy on the second argument. Meaning that if it meets `False` as the first argument, there is no need to calculate another arg. In `andM` we would like to keep the same behaviour, so the

```
Just False `andM` _
```

should be `Just False` :slightly_smiling_face: 


If you don't use the `>>=` operator, then the `Monad` constraint is too strong, and it is enough to specify only `Applicative`. However, with `Applicative` you are not able to implement a lazier version of `andM`.

Reply:
Understood. Thanks for a clear explanation 😊